### PR TITLE
Allow users to set a quality setting

### PR DESCRIPTION
**Pull request by Abbondanzo**
Abstracts asset loading to a simple loader interface so that users can modify the quality of the player. This lets us substitute full-quality models with boxes and simple assets for nonperformant phones and whatnot.

TODO Checklist:

- [x] Give the `LowQualityAssetLoader` some love (basic asset recreation)
- [x] Allow users to toggle the quality via a button in the player, next to fullscreen button
- [x] Support reloading assets during a play

Screenshots
Screen Shot 2020-02-27 at 8 14 23 AM
Screen Shot 2020-02-27 at 8 14 30 AM

### CODE REVIEW

File: src/loaders/impl/HighQualityAssetLoader.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  private setMaterials(body: BodyModel) {
+    body.scene.traverse(object => {
+      // @ts-ignore
+      if (object.isMesh) {
+        const mesh = object as Mesh
+        mesh.receiveShadow = true
+        mesh.castShadow = true
+
+        // Phong material is less physically accurate but has noticably better performance
```
Comment (Longi94) [inline src/loaders/impl/HighQualityAssetLoader.ts]: Might as well remove the usage of Phong material in HQ

File: src/builders/SceneBuilder.ts [hunk 2]
```diff
@@ -1,16 +1,65 @@
...
+const buildAssetManager = (loadingManager?: LoadingManager) => {
+  const gltfLoader = new GLTFLoader(loadingManager)
+  const dracoLoader = new DRACOLoader()
+  dracoLoader.setDecoderPath("/draco/")
+  gltfLoader.setDRACOLoader(dracoLoader)
+
+  const config = new RocketConfig({
+    gltfLoader,
+    loadingManager,
+    textureFormat: TextureFormat.PNG,
```
Comment (Longi94) [inline src/builders/SceneBuilder.ts]: for medium you can use `textureQuality: TextureQuality.LOW`, that will load lower resolution png files

Comment (Abbondanzo) [reply src/builders/SceneBuilder.ts]: Is this supported in a newer version? I'm not seeing support for it yet

**Comment by Longi94** [comment]
Wouldn't it make more sense to present to user with the option before loading things when the settings is not yet in local storage?

Also the quality button overlaps with the camera button in compact mode:

img

File: src/loaders/impl/LowQualityAssetLoader.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    const spherMaterial = new MeshStandardMaterial()
+    return new Mesh(sphereGeometry, spherMaterial)
+  }
+
+  async loadField(): Promise<Object3D> {
+    const geometry = new PlaneBufferGeometry(8192, 10240, 1, 1)
+    const material = new MeshBasicMaterial({ color: "#4CAF50" })
+    const fieldMesh = new Mesh(geometry, material)
+    fieldMesh.rotation.x = -Math.PI / 2
+    return fieldMesh
```
Comment (Longi94) [inline src/loaders/impl/LowQualityAssetLoader.ts]: I'd add something for the walls and ceiling

File: src/viewer/components/icons/QualityIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const QualityIcon = () => {
+  return (
+    <div>
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        height="24"
+        viewBox="0 0 24 24"
+        width="24"
+      >
+        <path d="M0 0h24v24H0z" fill="#fff" />
```
Comment (Longi94) [inline src/viewer/components/icons/QualityIcon.tsx]: white on transparent background for consistency?

**Pull request closed without merging**